### PR TITLE
[GlobalISel] Handle more commutable instructions in `commute_constant_to_rhs`

### PR DESCRIPTION
**Pull request by darkbuck**

### CODE REVIEW

**Review by arsenm — COMMENTED** [review]
It looks like most of these are untested, should add some tests

**Review by Pierre-vh — CHANGES_REQUESTED** [review]
Please add a MIR test for these combine. You can just have one function per opcode and then copy-paste it. Ideally each case should check both the easy case (just a constant on the lhs), but also the `G_CONSTANT_FOLD_BARRIER` case in `matchCommuteConstantToRHS`

**Comment by darkbuck** [comment]
@Pierre-vh could you review newly added tests?

**Review by arsenm — APPROVED** [review]

File: llvm/test/CodeGen/AArch64/GlobalISel/combine-commute-int-const-lhs.mir [hunk 1]
```diff
@@ -0,0 +1,456 @@
...
+tracksRegLiveness: true
+body:             |
+  bb.1:
+    liveins: $s0
+
+    ; CHECK-LABEL: name: umulfixsat
+    ; CHECK: liveins: $s0
+    ; CHECK-NEXT: {{  $}}
+    ; CHECK-NEXT: [[COPY:%[0-9]+]]:_(s32) = COPY $s0
+    ; CHECK-NEXT: %cst:_(s32) = G_CONSTANT i32 3
```
Comment (Pierre-vh) [inline llvm/test/CodeGen/AArch64/GlobalISel/combine-commute-int-const-lhs.mir:447]: Add a few test cases with the constant fold barrier instruction

Comment (darkbuck) [reply llvm/test/CodeGen/AArch64/GlobalISel/combine-commute-int-const-lhs.mir:447]: > Add a few test cases with the constant fold barrier instruction

Added several ones with different operand patterns other than the regular binary insn.

**Review by Pierre-vh — APPROVED** [review]

**Pull request merged**